distributedlog-core/src/main/java/com/twitter/distributedlog/BKDistributedLogNamespace.java

```diff
@@ -470,6 +478,23 @@ private BKDistributedLogNamespace(
                 new Object[] { clientId, regionId, bkdlConfig.isFederatedNamespace() });
     }
 
+    /**
+     * Validate the stream name.
+     *
+     * @param nameOfStream
+     *          name of stream
+     * @throws InvalidStreamNameException
+     */
+    private void validateName(String nameOfStream)
+            throws InvalidStreamNameException {
+        // validate the stream name
+        nsResolver.validateStreamName(nameOfStream);
+        if (isReservedStreamName(nameOfStream)) {
```

Comment:
shall we move the 'isReservedStreamName' validation to the resolver?

or shall we just deprecate using '.' for any component name in the path?